storage/src/test/java/org/apache/kafka/tiered/storage/integration/DisableRemoteLogOnTopicTest.java

```diff
@@ -0,0 +1,121 @@
+
+        builder
+                .createTopic(topicA, partitionCount, replicationFactor, maxBatchCountPerSegment, assignment,
+                        enableRemoteLogStorage)
+                // send records to partition 0
+                .expectSegmentToBeOffloaded(broker0, topicA, p0, 0, new KeyValueSpec("k0", "v0"))
+                .expectSegmentToBeOffloaded(broker0, topicA, p0, 1, new KeyValueSpec("k1", "v1"))
+                .expectEarliestLocalOffsetInLogDirectory(topicA, p0, 2L)
+                .produce(topicA, p0, new KeyValueSpec("k0", "v0"), new KeyValueSpec("k1", "v1"),
+                        new KeyValueSpec("k2", "v2"))
+                // disable remote log storage
+                .updateTopicConfig(topicA,
+                        Collections.singletonMap(TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, "false"),
+                        Collections.emptyList())
+
+                // make sure we can still consume from the beginning of the topic to read data from local and remote storage
+                .expectFetchFromTieredStorage(broker0, topicA, p0, 2)
+                .consume(topicA, p0, 0L, 3, 2)
+
+                // enable remote log storage
+                .updateTopicConfig(topicA,
+                        Collections.singletonMap(TopicConfig.REMOTE_LOG_STORAGE_ENABLE_CONFIG, "true"),
+                        Collections.emptyList())
+
+                // make sure the logs can be offloaded
+                .expectEarliestLocalOffsetInLogDirectory(topicA, p0, 3L)
+                .produce(topicA, p0, new KeyValueSpec("k3", "v3"))
+
+                // explicitly set delete policy as "retain" and disable remote storage
+                .updateTopicConfig(topicA,
+                        explictRetain,
+                        Collections.emptyList())
+                // make sure we can still consume from the beginning of the topic to read data from local and remote storage
+                .expectFetchFromTieredStorage(broker0, topicA, p0, 3)
+                .consume(topicA, p0, 0L, 4, 3)
+
+                // verify the remote retention policy is working.
+                // update retention size to 2, and we have 3 inactive segments with each size 1,
+                // so there will be 1 remote log segment get deleted
+                .updateTopicConfig(topicA,
+                        Collections.singletonMap(TopicConfig.RETENTION_BYTES_CONFIG, "2"),
+                        Collections.emptyList())
+                .expectDeletionInRemoteStorage(broker0, topicA, p0, DELETE_SEGMENT, 1)
+                .waitForRemoteLogSegmentDeletion(topicA)
+
+                // setting delete policy to "delete" and disable remote storage
+                .updateTopicConfig(topicA,
+                        deletePolicy,
+                        Collections.emptyList())
+                // make sure all remote data is deleted
+                .expectEmptyRemoteStorage(topicA, p0);
```

Comment:
Also, we can consume from the topic to verify that the messages in the local log segments are returned.

Reply:
Updated in https://github.com/apache/kafka/pull/16653/commits/5befac7404fae9486c628a0a48af763a99607845.